### ISSUE TITLE
Feature Request: Generic OIDC Support

### ISSUE THREAD

**Issue by mstrhakr** [Ylianst/MeshCentral#3814]
Hey, I would love to see an option for a generic/custom OIDC provider.  I'm using Authelia as an SSO provider and so far most services are taking to it well (not without a few hacks). I noticed (you told me) you have OIDC support already in the form of Azure AD, however I would need to be able to specify some extra things, like in the below example where I have a custom OIDC setup in nextcloud.

![custom oidc example nextcloud](https://user-images.githubusercontent.com/37352843/160471092-75ab304f-84f4-4aea-8e4d-fe4a00e8094d.png)

Here also is the config of Authelia showing that side of the equation.

![authelia oidc config example](https://user-images.githubusercontent.com/37352843/160471644-9c1ed007-4989-4dfe-b257-9b4388dabe03.png)

Let me know if you need more info or have any questions on this. I'd be happy to help.

EDIT: I wanted to give you this link the to OIDC setup page for Authelia, it helped me out in figuring this all out.
https://www.authelia.com/docs/configuration/identity-providers/oidc.html

**Comment by Ylianst** [comment]
Oh, Authelia looks very cool! I have some priority work I have to do for corporate customers... but this is a really good request. In general, I prefer having authentication integrated with other services, it make MeshCentral a lot easier to use.

**Comment by mstrhakr** [comment]
I'm happy to work with you on this to help in any way I can. Just let me know, I'll keep an eye on my notifications.

**Comment by mstrhakr** [comment]
Soooo, I'm impatient.. I'm also a budding programmer. I have a feeling you might see (and be annoyed with) where this is going.
Basically I forked the project, built the module in question and got it 90 percent working, however I'm missing something because I'm only getting redirected to the login screen instead of being logged in. I am going to keep working on this but figured I'd mention my progress in the hopes that you had some time to look over what I've done so far.
At the moment everything is local and if you respond wanting to see it all I'll get it uploaded to my github fork.

**Comment by mstrhakr** [comment]
A day later and it's working! I have no idea if what I've done is good enough for a pull request but if you want I will happily submit one. The only issue is I had to slightly modify a fork of the main 'passport-openidconnect' module since the passport dev stopped working on it, so there is no way to use npm to grab it (since I don't know how to upload to that)

This is the version I'm using with success, the only change I made was to add an email option in where it was ignored previously.
https://github.com/mstrhakr/passport-generic-oidc

And here is my fork of MeshCentral with the proposed changes.

Even if you never get see this I'll be happily using MeshCentral with Authelia integrated. 😄 
Next is to put it all together in a docker to replace my main setup, wish me luck :)
